lxd/storage/backend_lxd.go

```diff
@@ -213,7 +213,7 @@ func (b *lxdBackend) Delete(localOnly bool, op *operations.Operation) error {
 
 	// Delete the mountpoint.
 	path := shared.VarPath("storage-pools", b.name)
-	err := os.Remove(path)
+	err := os.RemoveAll(path)
```

Comment:
@monstermunchkin is it possible to fix the driver so that it does clean up properly, avoiding the need for this change?

Reply:
I will take another look.

Reply:
Indeed, we don't rely want to ever use `RemoveAll` in the new world. We should be aware of what's there, delete those bits properly and fail if the user has created things behind our back.

Reply:
The problem lies here: https://github.com/lxc/lxd/blob/master/lxd/storage/backend_lxd.go#L201-L212

For Ceph, it will not always call `Delete()` and therefore the pool directory will not always be wiped but contain the usual directory structure.

Reply:
@monstermunchkin ah, so its related to https://github.com/lxc/lxd/pull/6741#discussion_r375284954, its the reverse of that basically.